Add resource-config pages for microbatch configs (#6575)
dbt-labs/docs.getdbt.com: website/docs/reference/resource-configs/batch_size.md

---
title: "batch_size"
id: "batch-size"
sidebar_label: "batch_size"
resource_types: [models]

> description: "dbt uses `batch_size` to detrmine how large batches are when running a microbatch incremental model."

Reviewer: ```suggestion
description: "dbt uses `batch_size` to determine how large batches are when running a microbatch incremental model."
```

datatype: hour | day | month | year
---

> Available in dbt Cloud Versionless and dbt Core v1.9 and higher.

Reviewer: Would it be ok if we version the whole page? Any issues with that?

Author: I have no issue with that. The only reason I didn't was because I directly copied and then edited the `event_time.md` page

## Definition

> Set the `batch_size` either `hour`, `day`, `month`, or `year`. You can configure `batch_size` for a [model](/docs/build/models) in your `dbt_project.yml` file, property YAML file, or config block.

Reviewer: ```suggestion
The`batch_size` config determines how large batches are when running a microbatch. Accepted values are `hour`, `day`, `month`, or `year`. You can configure `batch_size` for a [model](/docs/build/models) in your `dbt_project.yml` file, property YAML file, or config block.
```

## Examples

> 

Reviewer: ```suggestion

The following examples set `day` as the `batch_size` for the `user_sessions` model.

```

<Tabs> 

> <TabItem value="model" label="Models">

Reviewer: @QMalcolm Do we consider this a "work in progress"? There's a single tab item in this section. Should each example be a tab?

<img width="906" alt="Screenshot 2024-12-03 at 12 11 52 AM" src="https://github.com/user-attachments/assets/a0d54e6c-916a-41ae-9bb5-e902918de725">

Author: More tabs are not expected. I just didn't know what the format should be if it's only applicable to `Models`, should it just be a sub heading instead?

> 

Reviewer: ```suggestion
The following examples set `day` as the `batch_size` for the `user_sessions` model.
```

> Here's an example in the `dbt_project.yml` file:

Reviewer: ```suggestion
Example of the `batch_size` config in the `dbt_project.yml` file:
```

<File name='dbt_project.yml'>

```yml
models:
  my_project:
    user_sessions:
      +batch_size: day
```
</File>

Example in a properties YAML file:

<File name='models/properties.yml'>

```yml
models:
  - name: user_sessions
    config:
      batch_size: day
```

</File>

Example in sql model config block:

<File name="models/user_sessions.sql">

```sql
{{ config(
    lookback='day
) }}
```

</File> 

> This setup sets `day` as the `batch_size` for the `user_sessions` model.

Reviewer: This might serve readers better upfront

> This setup sets `day` as the `batch_size` for the `user_sessions` model.

Reviewer: This might serve readers better upfront

```suggestion
```
